### ISSUE TITLE
'byte' format validation failure

### ISSUE THREAD

**Issue by bnacey-dil** [seriousme/fastify-openapi-glue#205]
I'm finding that the 'byte' format validation in the new 'oai-formats' will fail after the first validation.

It appears that the Regex object created for each invocation is retaining the 'lastIndex' from the previous validation.

This issue can be replicated by duplicating line 22 of test/test-oai-formats.js

**Comment by bnacey-dil** [comment]
Applying this patch to my project is meeting my needs. But I'm not sure this change keeps the original intent. 
```
diff --git a/node_modules/fastify-openapi-glue/lib/oai-formats.js b/node_modules/fastify-openapi-glue/lib/oai-formats.js
index 78f8de4..0f663bb 100644
--- a/node_modules/fastify-openapi-glue/lib/oai-formats.js
+++ b/node_modules/fastify-openapi-glue/lib/oai-formats.js
@@ -2,7 +2,7 @@ const Decimal = require('decimal.js-light');
 
 function validateByte() {
     // regExp borrowed from https://github.com/miguelmota/is-base64
-    var regex = new RegExp('^(?:[A-Za-z0-9+\\/]{4})*(?:[A-Za-z0-9+\\/]{2}==|[A-Za-z0-9+\/]{3}=)?$','gm');
+    var regex = new RegExp('^(?:[A-Za-z0-9+\\/]{4})*(?:[A-Za-z0-9+\\/]{2}==|[A-Za-z0-9+\/]{3}=)?$','m');
     return (value => value !== '' && regex.test(value))
 }
```

**Comment by seriousme** [comment]
It's a bug, thanks for reporting.

Removing the `g` would only check if the first line of the string is base64 and not the whole string.
The solution is to recreate the RexExp on every call as mentioned in: https://stackoverflow.com/questions/1520800/why-does-a-regexp-with-global-flag-give-wrong-results

I have created a patch.

Kind regards,
Hans

**Comment by seriousme** [comment]
Just published the patched version as 2.4.1 on npm.

Thanks again for reporting,
Hans

**Comment by bnacey-dil** [comment]
Thanks for looking into this so quickly. Version 2.4.1 is working as expected in my project.
